packages/teko/src/CMakeLists.txt

```diff
@@ -29,26 +29,46 @@ SET(HEADERS ${HEADERS}
 
 INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR})
 INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR}/NS)
-INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR}/Epetra)
 INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR}/Tpetra)
+
+IF(TEKO_HAVE_EPETRA)
+   INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR}/Epetra)
+ENDIF()
+
 IF (ML_ENABLE_TekoSmoothers)
    INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR}/mlutils)
 ENDIF()
 
 APPEND_GLOB(HEADERS
   *.hpp
   NS/*.hpp
-  Epetra/*.hpp
   Tpetra/*.hpp
   )
 
 APPEND_GLOB(SOURCES
   *.cpp
   NS/*.cpp
-  Epetra/*.cpp
   Tpetra/*.cpp
   )
 
+IF(NOT TEKO_HAVE_EPETRA)
+  LIST(REMOVE_ITEM SOURCES
+    ${CMAKE_CURRENT_SOURCE_DIR}/NS/Teko_ALOperator.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/Teko_DiagonalPreconditionerOp.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/Teko_DiagonalPreconditionerFactory.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/Teko_StratimikosFactory.cpp
+  )
+  LIST(REMOVE_ITEM HEADERS
+    ${CMAKE_CURRENT_SOURCE_DIR}/NS/Teko_ALOperator.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/Teko_DiagonalPreconditionerOp.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/Teko_DiagonalPreconditionerFactory.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/Teko_StratimikosFactory.hpp
```

Comment:
Why does the Stratimikos factory require Epetra?

Reply:
Yeah, I think that `Teko_StratimikosFactory` should be built also when `Epetra` is not enabled. If the PR is merged as is, I think it would break Albany.